File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/Location.java
package com.tregula.graedukacyjna.domain.database.entity;

import com.tregula.graedukacyjna.domain.database.entity.converter.LocationConverter;

import androidx.annotation.NonNull;
import androidx.room.TypeConverter;

public class Location {

    public final Double latitude;
    public final Double longitude;

    public Location(Double latitude, Double longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
    }

    @NonNull
    @Override
    public String toString() {
        return latitude.toString() +
                "," +
                longitude.toString();
    }

    public static class LocationTypeConverters {

        @TypeConverter
        public static Location fromString(String location) {
            return new LocationConverter().convert(location);
        }

        @TypeConverter
        public static String fromLocation(Location location) {
            return location.toString();
        }
    }
}

File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/Location.kt
package com.tregula.graedukacyjna.domain.database.entity

import com.tregula.graedukacyjna.domain.database.entity.converter.LocationConverter
import androidx.room.TypeConverter

data class Location(val latitude: Double, val longitude: Double) {

    override fun toString(): String {
        return latitude.toString() +
                "," +
                longitude.toString()
    }

    object LocationTypeConverters {

        @TypeConverter
        @JvmStatic
        fun fromString(location: String): Location {
            return LocationConverter().convert(location)
        }

        @TypeConverter
        @JvmStatic
        fun fromLocation(location: Location): String {
            return location.toString()
        }
    }
}
